### ISSUE TITLE
Optimize AptJob::searchPackageFiles()

### ISSUE THREAD

**Issue by sidt4** [PackageKit/PackageKit#658]
From https://gitlab.gnome.org/GNOME/gnome-software/-/issues/2254#note_1816171

Also, I think in `AptJob::searchPackageFiles()` we can break from the main `while` loop, once we have `g_strv_length(values)` entries in `vector<string>` packages. We expect a file to be found only in 1 package ( i.e `1 <-> 1` mapping). If not, it's a system error. I don't think we need to handle directories which can return multiple results.

**Comment by ximion** [comment]
> We expect a file to be found only in 1 package ( i.e `1 <-> 1` mapping). If not, it's a system error.

Not necessarily, as there are multiarch packages (e.g. amd64/arm64) as well as PPAs. I would expect this to be a very rare event though.

**Comment by sidt4** [comment]
Okay. In that case, this issue can be closed.
